### ISSUE TITLE
ENH: add false alarm probability to Lomb-Scargle

### ISSUE THREAD

**Issue by jakevdp** [astropy/astropy#5687]
I think this might be a useful enhancement, and could be accomplished in several dozen lines of code built on the current ``LombScargle`` class.

I'd imagine implementing a couple approaches:

- the "z-score" based on estimates of the number of independent frequencies
- the Baluev method, built on extreme value theory
- the bootstrap method

The API could look like:
```python
ls = LombScargle(t, mag, magerr)
freq, power = ls.autopower()

# find 95% and 99% levels in the periodogram
levels = ls.false_alarm_probability([95, 99], method='baluev')
```

Would this be a useful contribution? Any thoughts on API?

**Comment by pllim** [comment]
Maybe a better place to ask is the astropy user or dev mailing list. I don't use that myself, so even if I say yes, it means little.

**Comment by bsipocz** [comment]
I hope to go back to work a bit more on our light curves soon, and then if we ever end up rerunning LS, would probably use this feature, too (and also get rid of our python wrapped version of LS and use the astropy one...)

**Comment by pllim** [comment]
Haven't seen a PR on this, so removing milestone.

**Comment by jakevdp** [comment]
Closed by #6488
